qutebrowser/browser/greasemonkey.py

```diff
@@ -119,9 +119,9 @@ def code(self):
         template = jinja.js_environment.get_template('greasemonkey_wrapper.js')
         return template.render(
             scriptName=javascript.string_escape(
-                "/".join([self.namespace or '', self.name])),
+                "/".join([self.namespace or '', self.name or ''])),
             scriptInfo=self._meta_json(),
-            scriptMeta=javascript.string_escape(self.script_meta),
+            scriptMeta=javascript.string_escape(self.script_meta or ''),
```

Comment:
@toofar "it is perfectly legal for a script not to have a metadata block" - is it? https://wiki.greasespot.net/Metadata doesn't say much unfortunately, but I'd expect the metadata block to be mandatory?

Reply:
Hmm, I was sure it was and tried adding js-only scripts to tampermonkey and violetmonkey and they both said it was invalid. I don't see why it would be required though when all of the keys are optional.
> If no name is provided, it will be derived from the file name.
> If no include rule is provided, @include * is assumed.